modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureClientProvider.java

```diff
@@ -238,13 +238,8 @@ protected void doStop() {
     @Override
     protected void doClose() throws IOException {}
 
-    private static final class SuccessfulRequestTracker implements HttpPipelinePolicy {
-        private final BiConsumer<String, URL> onSuccessfulRequest;
-        private final Logger logger = LogManager.getLogger(SuccessfulRequestTracker.class);
-
-        private SuccessfulRequestTracker(BiConsumer<String, URL> onSuccessfulRequest) {
-            this.onSuccessfulRequest = onSuccessfulRequest;
-        }
+    private record SuccessfulRequestTracker(BiConsumer<String, URL> onSuccessfulRequest) implements HttpPipelinePolicy {
```

Comment:
Does it make sense for this class to be a record? The field isn't exposed, its not used in a hashtable...

Reply:
I guess I just auto-refactored that because it was shorter now. I had that same discussion in another PR though where I asked that question from your perspective :D Tbh. I don't know, in the other PR we came to the conclusion that a record is less verbose and the fact that we have hashCode and equals implemented is mostly irrelevant, so less code == good -> record ... But happy to revert that, it's kind of a random thing.

Reply:
I think something should only be a record if its actually a POJO being passed around. Here it's a single field, but with a method doing something. So adding record semantics here is confusing.

Reply:
++ fine by me :) reverted